dora/core/client/fs/src/main/java/alluxio/client/file/dora/LocalWorkerPolicy.java

```diff
@@ -0,0 +1,43 @@
+package alluxio.client.file.dora;
+
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.util.network.NetworkAddressUtils;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * A policy where a client will talk to a local worker.
+ * The most common use case is for testing like StressWorkerBench, where we create clients
+ * (with job service) and simulate I/O workload to workers.
+ * If all clients will talk to a local worker, the test creates balanced stress on each worker
+ * and the performance measured is I/O with perfect data locality.
+ */
+public class LocalWorkerPolicy implements WorkerLocationPolicy {
+  private final AlluxioConfiguration mConf;
+  public LocalWorkerPolicy(AlluxioConfiguration conf) {
+    mConf = conf;
+  }
+
+  @Override
+  public List<BlockWorkerInfo> getPreferredWorkers(
+      List<BlockWorkerInfo> blockWorkerInfos, String fileId, int count) {
+    String userHostname = NetworkAddressUtils.getClientHostName(mConf);
+    // Find the worker matching in hostname
+    // TODO(jiacheng): domain socket is not considered here
+    BlockWorkerInfo localWorker = null;
+    for (BlockWorkerInfo worker : blockWorkerInfos) {
+      if (worker.getNetAddress().getHost().equals(userHostname)) {
+        localWorker = worker;
+        break;
+      }
+    }
+    if (localWorker != null) {
+      return ImmutableList.of(localWorker);
+    } else {
+      return ImmutableList.of();
```

Comment:
let's think again about throwing an exception here?

Reply:
Considering the policy implementation itself, I think it is okay to return an empty list. Then we can leave the responsibility of checking for an empty list to the user of `LocalWorkerPolicy`.

Reply:
+1 on `LocalWorkerPerferredPolicy`. Let's rename this policy and in case no local worker can be found, return any one of the remote workers.

Reply:
That's the problem, if an "any worker" is returned, this policy is not deterministic! If in a test you want a local worker but got an "any worker" by mis-configuration, that's not what you want either.